pkg/proxy/proxy.go

```diff
@@ -146,6 +146,114 @@ func (p *SprayProxy) HandleProxy(c *gin.Context) {
 	c.String(http.StatusOK, "proxied")
 }
 
+func (p *SprayProxy) HandleProxyEndpoint(c *gin.Context) {
+	// currently not distinguishing between requests we can parse and those we cannot parse
+	metrics.IncInboundCount()
+	errors := []error{}
+	zapCommonFields := []zapcore.Field{
+		zap.String("method", c.Request.Method),
+		zap.String("path", c.Request.URL.Path),
+		zap.String("query", c.Request.URL.RawQuery),
+		zap.Bool("insecure-tls", p.insecureTLS),
+	}
+	// Read in body from incoming request
+	buf := &bytes.Buffer{}
+	_, err := buf.ReadFrom(c.Request.Body)
+	defer c.Request.Body.Close()
+	if err != nil {
+		c.String(http.StatusRequestEntityTooLarge, "too large: %v", err)
+		p.logger.Error("request body too large", zapCommonFields...)
+		return
+	}
+	body := buf.Bytes()
+
+	client := &http.Client{}
+	if p.insecureTLS {
+		client.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		}
+	}
+
+	for _, backend := range p.backends() {
+		backendURL, err := url.Parse(backend)
+		if err != nil {
+			p.logger.Error("failed to parse backend "+err.Error(), zapCommonFields...)
+			continue
+		}
+		copy := c.Copy()
+		newURL := copy.Request.URL
+		newURL.Host = backendURL.Host
+		newURL.Scheme = backendURL.Scheme
+
+		// remove /proxy from the copied backend URL
+		newBackendURL := fmt.Sprintf("%s://%s", backendURL.Scheme, backendURL.Host)
+
+		// zap always append and does not override field entries, so we create
+		// per backend list of fields
+		zapBackendFields := append(zapCommonFields, zap.String("backend", newBackendURL))
+		newRequest, err := http.NewRequest(copy.Request.Method, newBackendURL, bytes.NewReader(body))
```

Comment:
This will not work - in fact this uncovers gaps in our testing:

- Request headers need to be preserved. These convey important information, especially from GitHub. For example, webhook authentication via the GitHub signature header will break once the request is forwarded to PaC on the member cluster.
- Paths after `/proxy/` need to be preserved.

The better approach is to update `newURL.Path` by using https://pkg.go.dev/strings#TrimPrefix. Here's a quick go playground that demonstrates how I think it can work: https://go.dev/play/p/W38wK--GSDk

Reply:
> Request headers need to be preserved. These convey important information, especially from GitHub. For example, webhook authentication via the GitHub signature header will break once the request is forwarded to PaC on the member cluster.

ohh yes as we have added the request-ids it will break those. 
Fixed. I have modified the test as well accordingly.